src/main/java/uk/gov/hmcts/reform/civil/handler/tasks/BaseExternalTaskHandler.java

```diff
@@ -55,25 +55,27 @@ default void execute(ExternalTask externalTask, ExternalTaskService externalTask
             log.error("External task '{}' errored  with processInstanceId '{}'",
                       topicName, processInstanceId, e
             );
-            handleFailureNoRetryable(externalTask, externalTaskService, e);
+            handleFailureNotRetryable(externalTask, externalTaskService, e);
         } catch (Exception e) {
             log.error("External task before handleFailure '{}' errored  with processInstanceId '{}'",
                       topicName, processInstanceId, e
             );
             handleFailure(externalTask, externalTaskService, e);
         }
+
+        if (handleTaskSucceeded) {
+            completeTask(externalTask, externalTaskService);
+        }
     }
 
+    //Total possible waiting time 16 minutes - if changing this, change lockDuration in ExternalTaskListenerConfiguration
     @Retryable(value = CompleteTaskException.class, maxAttempts = 3, backoff = @Backoff(delay = 60000, multiplier = 15))
     default void completeTask(ExternalTask externalTask, ExternalTaskService externalTaskService) throws CompleteTaskException {
         String topicName = externalTask.getTopicName();
         String processInstanceId = externalTask.getProcessInstanceId();
 
         try {
-            ofNullable(getVariableMap()).ifPresentOrElse(
```

Comment:
There is no need for this null check as you can just call the the complete method passing the variable and if it is null or not it will be handled later if you look into the complete method you will see it.